### ISSUE TITLE
Emoticons in cache descriptions are extremely small.

### ISSUE THREAD

**Issue by eliannelavoie** [cgeo/cgeo#13648]
### Describe your problem!

Old school emoticons (not emojis) in cache descriptions are much smaller than the text.

![Screenshot_20221112-203858](https://user-images.githubusercontent.com/13744678/201501492-5d5d86e8-200d-4455-ba33-40bdd5971a1f.png)

### How to reproduce?

Open the cache with Geocode GC62QE5.

Go to the description tab.

### Reproducible

Yes

### c:geo Version

2022.10.17

### System information

```text
## System information

c:geo version: 2022.10.17

Device:
-------
- Device type: Pixel 6 Pro (raven, google)
- Available processors: 8
- Android version: 13
- Android build: TP1A.221005.002
- Screen resolution: 1440x2899px (380x765dp)
- Pixel density: 3.40875
- System font scale: 0.9 / used scale: 0.9
- Sailfish OS detected: false
- Google Play services: enabled - 22.43.12 (190400-483592595)

Sensor and location:
-------
- Low power mode: inactive
- Compass capabilities: yes
- Rotation vector sensor: present
- Orientation sensor: present
- Magnetometer & Accelerometer sensor: present
- Direction sensor used: rotation vector

Program settings:
-------
- Settings: v9, Count:82
- Set language: en_CA (system default)
- System date format: y-MM-dd
- Time zone: EST
- Debug mode active: no
- Last backup: never
- Routing mode: Walk
- Live map mode: true
- OSM multi-threading: false / threads: 1
- Map: Quebec (Offline)
  - Id: cgeo.geocaching.maps.mapsforge.MapsforgeMapProvider$OfflineMapSource:primary:Geocaching/maps/quebec.map
  - Atts: Map data (c) OpenStreetMap contributors
  - Theme: none

Filters:
-------
- Hide waypoints: parking 
- LIVE: None ([:inconclusive=false:advanced=false])
- OFFLINE: None ([:inconclusive=false:advanced=false])

Stored Filters (#0):

Services:
-------
- Geocaching sites enabled:
   geocaching.com: Logged in (Login OK) / BASIC
- Geocaching.com date format: M/d/yyyy
- Geocaching.com website language: en-US
- Last successful login on geocaching.com: 2022-11-12 18-33
- Routing: internal / BRouter installed: false
- Installed c:geo plugins:  none

Permissions & paths:
-------
- Fine location permission: granted
- Write external storage permission: granted
- System internal c:geo dir: /data/user/0/cgeo.geocaching (4.2 GB free) v3 internal isDir(7 entries)
- Legacy User storage c:geo dir: /data/user/0/cgeo.geocaching (4.2 GB free) v3 internal isDir(7 entries)
- Geocache data: /storage/emulated/0/Android/data/cgeo.geocaching/files/GeocacheData (4.2 GB free) v3 external non-removable isDir(70 entries)
- Internal theme sync (is turned off): /data/user/0/cgeo.geocaching/MapThemeData (4.2 GB free) v3 internal isDir(0 entries)
- Public Folders: #12
  - BASE: /Geocaching (User-Defined)[/Geocaching[DOCUMENT#0:p-content://com.android.externalstorage.documents/tree/primary%3AGeocaching::]] (Uri: content://com.android.externalstorage.documents/tree/primary%3AGeocaching/document/primary%3AGeocaching, Av:true, files:>=2, dirs:>=9, totalFileSize:>=524.9 MB, free space: 4.2 GB, files on device: 13278717)
  - OFFLINE_MAPS: /Geocaching/maps (Default)[/Geocaching/maps[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3AGeocaching::/maps]] (Uri: content://com.android.externalstorage.documents/tree/primary%3AGeocaching/document/primary%3AGeocaching%2Fmaps, Av:true, files:2, dirs:2, totalFileSize:524.9 MB, free space: 4.2 GB, files on device: 13278717)
  - OFFLINE_MAP_THEMES: /Geocaching/maps/_themes (Default)[/Geocaching/maps/_themes[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/primary%3AGeocaching::/maps/_themes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3AGeocaching/document/primary%3AGeocaching%2Fmaps%2F_themes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 4.2 GB, files on device: 13278717)
  - OFFLINE_MAP_SHADING: /Geocaching/maps/_hgt (Default)[/Geocaching/maps/_hgt[PERSISTABLE_FOLDER(OFFLINE_MAPS)#1:p-content://com.android.externalstorage.documents/tree/primary%3AGeocaching::/maps/_hgt]] (Uri: content://com.android.externalstorage.documents/tree/primary%3AGeocaching/document/primary%3AGeocaching%2Fmaps%2F_hgt, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 4.2 GB, files on device: 13278717)
  - LOGFILES: /Geocaching/logfiles (Default)[/Geocaching/logfiles[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3AGeocaching::/logfiles]] (Uri: content://com.android.externalstorage.documents/tree/primary%3AGeocaching/document/primary%3AGeocaching%2Flogfiles, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 4.2 GB, files on device: 13278717)
  - GPX: /Geocaching/gpx (Default)[/Geocaching/gpx[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3AGeocaching::/gpx]] (Uri: content://com.android.externalstorage.documents/tree/primary%3AGeocaching/document/primary%3AGeocaching%2Fgpx, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 4.2 GB, files on device: 13278717)
  - BACKUP: /Geocaching/backup (Default)[/Geocaching/backup[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3AGeocaching::/backup]] (Uri: content://com.android.externalstorage.documents/tree/primary%3AGeocaching/document/primary%3AGeocaching%2Fbackup, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 4.2 GB, files on device: 13278717)
  - FIELD_NOTES: /Geocaching/field-notes (Default)[/Geocaching/field-notes[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3AGeocaching::/field-notes]] (Uri: content://com.android.externalstorage.documents/tree/primary%3AGeocaching/document/primary%3AGeocaching%2Ffield-notes, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 4.2 GB, files on device: 13278717)
  - SPOILER_IMAGES: /Geocaching/GeocachePhotos (Default)[/Geocaching/GeocachePhotos[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3AGeocaching::/GeocachePhotos]] (Uri: content://com.android.externalstorage.documents/tree/primary%3AGeocaching/document/primary%3AGeocaching%2FGeocachePhotos, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 4.2 GB, files on device: 13278717)
  - ROUTING_BASE: /Geocaching/routing (Default)[/Geocaching/routing[PERSISTABLE_FOLDER(BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3AGeocaching::/routing]] (Uri: content://com.android.externalstorage.documents/tree/primary%3AGeocaching/document/primary%3AGeocaching%2Frouting, Av:true, files:7, dirs:1, totalFileSize:81.7 KB, free space: 4.2 GB, files on device: 13278717)
  - ROUTING_TILES: /Geocaching/routing/segments4 (Default)[/Geocaching/routing/segments4[PERSISTABLE_FOLDER(ROUTING_BASE)#1:p-content://com.android.externalstorage.documents/tree/primary%3AGeocaching::/routing/segments4]] (Uri: content://com.android.externalstorage.documents/tree/primary%3AGeocaching/document/primary%3AGeocaching%2Frouting%2Fsegments4, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 4.2 GB, files on device: 13278717)
  - TEST_FOLDER: [Legacy]/data/user/0/cgeo.geocaching/files/unittest (Default)[/data/user/0/cgeo.geocaching/files/unittest[FILE#1:p-file:///data/user/0/cgeo.geocaching/files::/unittest]] (Uri: file:///data/user/0/cgeo.geocaching/files/unittest, Av:true, files:0, dirs:0, totalFileSize:0 B, free space: 4.2 GB, files on device: -1)
- Map render theme path: 
- PersistedDocumentUris: #1
- UNUSED_TRACK: null
- Persisted Uri Permissions: #1
  - content://com.android.externalstorage.documents/tree/primary%3AGeocaching (Nov 8, 2:16 a.m.):RW
- Database: /data/user/0/cgeo.geocaching/databases/data (v99, Size:2.1 MB) on system internal storage

--- End of system information ---
```

**Comment by bekuno** [comment]
@eliannelavoie 
Thanks for the report.
The given cache geocode does not fit to your screen shot.
Beside that, the described problem is reproducible. The current rendering of emoji in the cache description is not optimal.
E.g. for the GC62QE5:
On the website:
![20221113_085923 1](https://user-images.githubusercontent.com/32555742/201529709-a22e70f9-1e21-4669-9788-cf82756f685a.jpg)
In c:geo:
![Screenshot_20221113-085728_1](https://user-images.githubusercontent.com/32555742/201529690-60eebb91-71b4-418a-aed2-a50452d13efc.jpg)

@eliannelavoie Do you want to (try to) solve this by yourself?

**Comment by moving-bits** [comment]
Those emojis are images, not Unicodes characters, thus they are not scaled by Android's text system.

One would need to identify the images and try to scale them. (Maybe by checking whether the actual height of an image is smaller than text height. Not sure about possible side effects, though.)

**Comment by bekuno** [comment]
> Those emojis are images, not Unicodes characters, thus they are not scaled by Android's text system.
> 
> One would need to identify the images and try to scale them. (Maybe by checking whether the actual height of an image is smaller than text height. Not sure about possible side effects, though.)

Correct. The scaling is a html parameter:
`<img alt="devil" src="[https://www.geocaching.com/static/js/CKEditor/4.4.0/plugins/smiley/images/devil_smile.png](view-source:https://www.geocaching.com/static/js/CKEditor/4.4.0/plugins/smiley/images/devil_smile.png)" title="devil" style="height:23px;width:23px;">`

**Comment by fm-sys** [comment]
Side note: Inside the logs emoticons are scaled properly... However this implementation of emoticon image handling would conflict with our current image handling inside the listing.

**Comment by bekuno** [comment]
> Side note: Inside the logs emoticons are scaled properly... However this implementation of emoticon image handling would conflict with our current image handling inside the listing.

@fm-sys In this situation we do not have a emoticon, here it is a plain *.png image.

**Comment by fm-sys** [comment]
> > Side note: Inside the logs emoticons are scaled properly... However this implementation of emoticon image handling would conflict with our current image handling inside the listing.
> 
> @fm-sys In this situation we do not have a emoticon, here it is a plain *.png image.

I am aware of this, but there actually _can_ be *.png images inside the log text as well... 

See https://github.com/cgeo/cgeo/blob/master/main/src/cgeo/geocaching/network/SmileyImage.java

![Screenshot_20221113_230323_cgeo geocaching](https://user-images.githubusercontent.com/64581222/201546880-9975c16f-c89e-4299-aa49-a36d4a592f6a.png)

https://coord.info/GL18Q8TFP

**Comment by eliannelavoie** [comment]
Geocache.java already has some logic to separate static images from non-static ones:

https://github.com/cgeo/cgeo/blob/ee0652048b8f7ce2f31f72f88ba85ce28913b86a/main/src/cgeo/geocaching/models/Geocache.java#L2161-L2177

Something similar could be used to apply the scaling only to images that are emoticons.

The urls to emoticons look like this: `https://www.geocaching.com/static/js/CKEditor/4.4.0/plugins/smiley/images/devil_smile.png`

I think filtering for "/smiley/images" would work.

@bekuno Sure, I'll try to make a fix. I'm getting acquainted with the project right now!

**Comment by bekuno** [comment]
> @bekuno Sure, I'll try to make a fix. I'm getting acquainted with the project right now!

That fine.
Dont hesitate to ask if you need help.

Maybe a more detailed filter would be better, something like `js/CKEditor/*/plugins/smiley/images`.

**Comment by moving-bits** [comment]
Resolved in current nightlies with PR #14549, so let's close here
